application/features/method.feature

```diff
@@ -0,0 +1,85 @@
+
+  Scenario: Retrieve Method records for a User without any Method records
+    When I request "/user/123/method" be retrieved
+    Then the response status code should be 200
+    And I should receive 0 records
+
+  Scenario: Retrieve Method records for a User with 1 Method record
+    Given user with employee id 123 has a verified Method
+    When I request "/user/123/method" be retrieved
+    Then the response status code should be 200
+    And I should receive 1 record
+
+  Scenario: Retrieve a Method for a User with 1 verified and 1 unverified Method
+    Given user with employee id 123 has a verified Method
+    And user with employee id 123 has an unverified Method
+    When I request "/user/123/method" be retrieved
+    Then the response status code should be 200
+    And I should receive 1 record
+
+  Scenario: Retrieve a specific Method record
+    Given user with employee id 123 has a verified Method
+    When I send a "GET" to "/method/{uid}" with a valid uid
+    Then the response status code should be 200
+    And the following data is returned:
+      | property      | value                 |
+      | value         | verified@example.com  |
+
+  Scenario: Create a new Method record
+    When I provide the following valid data:
+      | property     | value                 |
+      | employee_id  | 123                   |
+      | value        | user123@example.org   |
+    And I request "/method/create" be created
+    Then the response status code should be 200
+    And the following data is returned:
+      | property      | value                 |
+      | value         | user123@example.org   |
+    And a method record exists with a value of "user123@example.org"
+    And the following method data should be stored:
+      | property              | value                 |
+      | value                 | user123@example.org   |
+      | verified              | 0                     |
+      | verification_attempts | 1                     |
+#    And verification_code should be stored as ?
+#    And verification_expires should be stored as ? UTC
+#    And created should be stored as now UTC
+#    And an email is sent to "user123@example.org"
+
+### TODO: Try to create a method using the primary (or spouse/supervisor) address
+
+ Scenario: Resend a method verification
+   Given user with employee id 123 has an unverified Method
+   When I send a "PUT" to "/method/{uid}/resend" with a valid uid
+   Then the response status code should be 200
+   And a method record exists with a value of "unverified@example.com"
+   And the following method data should be stored:
+     | property              | value                 |
+     | verified              | 0                     |
+#   And an email is sent to "unverified@example.com"
+
+ Scenario: Verify a Method
+   Given user with employee id 123 has an unverified Method
+   When I send a "PUT" to "/method/{uid}/verify" with a valid uid
```

Comment:
It seems unclear in this test how the method was verified. Isn't a `verify` call required to provide the correct `verification_code`?

Reply:
Not sure I understand your question, but it's probably a poorly written `feature`. The verify call in line 69 does the verify. The `code` is retained in the context definition of the previous line and then sent by the verify call. Do you have any ideas for a more clear feature description?

Reply:
I think this would be clearer:
```suggestion
   When I send the correct code to verify that Method
```

Part of the problem, I think, is that the `Given user with employee id 123 has an unverified Method` step is doing more than it claims to: it's also inserting the correct `verification_code` into the request body for whatever request is made next.

Instead, I'd recommend we wait to add that code to the request until we're in a step that specifically mentions it. As an example, we could remove the ` $this->iChangeThe('code', $method->verification_code)` line from `MethodContext->userHasAMethod(...)` and instead use something like the following for `When I send the correct code to verify that Method`:
```php
    /**
     * @When I send the correct code to verify that Method
     */
    public function iSendTheCorrectCodeToVerifyThatMethod()
    {
        $method = Method::findOne(['uid' => $this->tempUid]);
        $this->iChangeThe('code', $method->verification_code);
        $this->iSendAToWithAValidUid('PUT', '/method/{uid}/verify');
    }
```

Reply:
I implemented your suggestions. Thank you.